astropy/time/formats.py

```diff
@@ -247,8 +247,13 @@ def cache(self):
     def _check_val_type(self, val1, val2):
         """Input value validation, typically overridden by derived classes"""
         # val1 cannot contain nan, but val2 can contain nan
+        isfinite1 = np.isfinite(val1)
+        if val1.size > 1:  # Calling .all() on a scalar is surprisingly slow
+            isfinite1 = isfinite1.all()  # Note: arr.all() about 3x faster than np.all(arr)
+        elif val1.size == 0:
```

Comment:
to keep optimum speed for the `size==1` case, maybe just use `val1.size != 1` above? (I know we're talking ns now, but it also keeps the code simpler, and the result doesn't matter below)

Reply:
I'm going to leave it since (for me anyway) even though the code is longer, it is more explicit in the different cases and easier to read.  Otherwise one needs to mentally process the two separate cases.  Another `if` statement is indeed in the nanosec.